anvil/src/genesis.rs

```diff
@@ -0,0 +1,246 @@
+//! Bindings for geth's `genesis.json` format
+use crate::revm::AccountInfo;
+use ethers::types::{serde_helpers::*, Address, Bytes, H256, U256};
+use forge::revm::{Bytecode, Env, KECCAK_EMPTY};
+use foundry_common::errors::FsPathError;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, path::Path};
+
+/// Genesis specifies the header fields, state of a genesis block. It also defines hard fork
+/// switch-over blocks through the chain configuration See also: <https://github.com/ethereum/go-ethereum/blob/0ce494b60cd00d70f1f9f2dd0b9bfbd76204168a/core/genesis.go#L47-L66>
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Genesis {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub config: Option<Config>,
+    #[serde(
+        default,
+        deserialize_with = "deserialize_stringified_u64_opt",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub nonce: Option<u64>,
+    #[serde(
+        default,
+        deserialize_with = "deserialize_stringified_u64_opt",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub timestamp: Option<u64>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub extra_data: Option<Bytes>,
+    #[serde(deserialize_with = "deserialize_stringified_u64")]
+    pub gas_limit: u64,
+    #[serde(deserialize_with = "deserialize_stringified_u64")]
+    pub difficulty: u64,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub mix_hash: Option<H256>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub coinbase: Option<Address>,
+    #[serde(default)]
+    pub alloc: Alloc,
+    #[serde(
+        default,
+        deserialize_with = "deserialize_stringified_u64_opt",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub number: Option<u64>,
+    #[serde(
+        default,
+        deserialize_with = "deserialize_stringified_u64_opt",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub gas_used: Option<u64>,
```

Comment:
i haven't seen `gas_used` in any available reference files. did u use some particular spec for `genesis.json`?

Reply:
yeh these are 1:1 bindings for geth's format,ported go structs, unsure where this is used tbh